src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -335,11 +353,41 @@
     (->> (map format-task info-from-jruby)
          (middleware-utils/json-response 200))))
 
+(schema/defn ^:always-validate
+  describe-task :- TaskDetails
+  [task-data :- TaskData
+   env-name :- schema/Str
+   module-name :- schema/Str]
+  (let [?metadata (try (some-> task-data :metadata_file slurp json/decode)
+                   (catch Exception e
+                     nil))
+        describe-file (fn [file]
+                        (let [size (fs/size file)
+                              sha256 (ks/file->sha256 (io/file file))
+                              base-name (fs/base-name file)]
+                          {:filename base-name
+                           :sha256 sha256
+                           :size_bytes size
+                           :uri {:path (str "/puppet/v3/file_content/tasks/" module-name "/" base-name)
+                                 :params {:environment env-name}}}))]
+    {:metadata (or ?metadata {})
```

Comment:
Seems like you could do the `{}` defaulting in your catch above instead of here and not need the `or` here.

Reply:
Yeah, that would be possible. I kind of like doing it here because it's a little easier to see that there's a default `{}` without all the clutter from the try/catch stuff.

Reply:
Yea I'm not sure which is clearer, so I'm fine either way.